src/inference/dev_api/openvino/itensor.hpp

```diff
@@ -0,0 +1,139 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief A header file for Tensor implementation
+ *
+ * @file itensor.hpp
+ */
+#pragma once
+
+#include "ie_blob.h"
+#include "openvino/core/coordinate.hpp"
+#include "openvino/core/shape.hpp"
+#include "openvino/core/type/element_type.hpp"
+#include "openvino/runtime/allocator.hpp"
+#include "openvino/runtime/common.hpp"
+
+namespace ov {
+
+/**
+ * @brief This class represents a universal container in the OpenVINO
+ *
+ * @note Each ITensor implementation must be derived from this ITensor class directly or indirectly
+ */
+class OPENVINO_API ITensor : public std::enable_shared_from_this<ITensor> {
+public:
+    /**
+     * @brief A smart pointer containing ITensor object
+     */
+    using Ptr = std::shared_ptr<ITensor>;
+
+    /**
+     * @brief Set new shape for tensor, deallocate/allocate if new total size is bigger than previous one.
+     * @note Memory allocation may happen
+     * @param shape A new shape
+     */
+    virtual void set_shape(const ov::Shape& shape);
+
+    /**
+     * @return A tensor element type
+     */
+    virtual element::Type get_element_type() const;
```

Comment:
The return value will most probably always refer to the constexpr objects defined in element_type.hpp so is there any reason NOT to return by const ref?